decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] in the Decorator class you should also include the attribute accessor for the `@nameable`.

Reply:
Thanks for the quick answer @mikethreels 
Got it, I will add these